### ISSUE TITLE
export `Text` conflicting with Julia Base export

### ISSUE THREAD

**Issue by Moelf** [MakieOrg/Makie.jl#3761]
```julia
julia> using CairoMakie

julia> Text
WARNING: both CairoMakie and Base export "Text"; uses of it in module Main must be qualified
ERROR: UndefVarError: `Text` not defined
```

**Comment by asinghvi17** [comment]
I don't think this can be avoided since that's defined all the way down in MakieCore, and is the type of the `text` recipe...

**Comment by Moelf** [comment]
Why export it?

**Comment by jkrumbiegel** [comment]
The export statement is baked into the `@recipe` macro

**Comment by asinghvi17** [comment]
In all fairness,
>   │  Text is currently exported to maintain backwards compatibility, but this export is deprecated. It is recommended to use this type as Docs.Text or to explicitly import it from Docs.

**Comment by Moelf** [comment]
yeah but as long as it's exported, Makie is doing the job of "breacking backwards compatibility" for no clear gain, we should just wait for Julia 2.0 if we want to export Text from Makie

**Comment by asinghvi17** [comment]
In this case I think Simon's `Minimal` idea could work - but I don't see how to suppress the export of `Text` from the macro without special casing it within that macro which is a bit of a code smell.  

Maybe a keyword `export=true` and then we set that to false and explicitly export everything but `Text` for that recipe?

**Comment by Moelf** [comment]
do we change what that macro expands to often? if not, we can "paste" the result of that macro and remove the export part
